### PR TITLE
Update check-before-deployment.md

### PR DESCRIPTION
**Pull request by Edwinmaster**
add vm.min_free_kbytes  description

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):https://docs.kernel.org/admin-guide/sysctl/vm.html#:~:text=a%20memory%20failure.-,min_free_kbytes,based%20proportionally%20on%20its%20size.

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: check-before-deployment.md [hunk 1]
```diff
@@ -626,9 +626,15 @@ sudo systemctl enable ntpd.service
     echo "net.ipv4.tcp_tw_recycle = 0">> /etc/sysctl.conf
     echo "net.ipv4.tcp_syncookies = 0">> /etc/sysctl.conf
     echo "vm.overcommit_memory = 1">> /etc/sysctl.conf
+    echo "vm.min_free_kbytes = 1048576">> /etc/sysctl.conf
     sysctl -p
     ```
 
+> **注意：**
+>
+> - vm.min_free_kbytes 是 Linux 内核用于控制系统预留的最少空闲内存，单位 KB。
```
Comment (Connor1996) [inline check-before-deployment.md:635]: 最好解释下原因，以及这个参数是如何影响的。我感觉一直设置 1GB 会不会太多

**Review by Connor1996 — APPROVED** [review]
LGTM

File: check-before-deployment.md [hunk 2]
```diff
@@ -626,9 +626,17 @@ sudo systemctl enable ntpd.service
...
+> - vm.min_free_kbytes 影响内存回收机制，设置过大会导致可用内存变少，设置过小可能因为申请内存过快超过后台回收的速度而导致进行直接内存回收引起 alloc stall。
+> - 最小设置 1 GB，当开启 numa 情况下建议 numa 个数 * 1 GB。
+> - 小规格服务器如系统内存小于 16 GB，保持系统默认值即可。
```
Comment (qiancai) [inline check-before-deployment.md:638]:
```suggestion
> - `vm.min_free_kbytes` 是 Linux 内核的一个参数，用于控制系统预留的最小空闲内存量，单位为 KiB。
> - `vm.min_free_kbytes` 的设置会影响内存回收机制。设置得过大，会导致可用内存变少，设置得过小，可能会导致内存的申请速度超过后台的回收速度，进而导致内存回收并引起内存分配延迟。
> - 建议将 `vm.min_free_kbytes` 最小设置为 1048576 KiB（即 1 GiB）。如果[安装了 NUMA](/check-before-deployment.md#安装-numactl-工具)，建议设置为 NUMA 节点个数 * 1048576 KiB。
> - 对于内存小于 16 GiB 的小规格服务器，保持系统默认的 `vm.min_free_kbytes` 值即可。
```

File: check-before-deployment.md [hunk 3]
```diff
@@ -626,9 +626,17 @@ sudo systemctl enable ntpd.service
...
     echo "vm.overcommit_memory = 1">> /etc/sysctl.conf
+    echo "vm.min_free_kbytes = 1048576">> /etc/sysctl.conf
     sysctl -p
     ```
 
+> **注意：**
+>
+> - `vm.min_free_kbytes` 是 Linux 内核的一个参数，用于控制系统预留的最小空闲内存量，单位为 KiB。
+> - `vm.min_free_kbytes` 的设置会影响内存回收机制。设置得过大，会导致可用内存变少，设置得过小，可能会导致内存的申请速度超过后台的回收速度，进而导致内存回收并引起内存分配延迟。
+> - 建议将 `vm.min_free_kbytes` 最小设置为 1048576 KiB（即 1 GiB）。如果[安装了 NUMA](/check-before-deployment.md#安装-numactl-工具)，建议设置为 `NUMA 节点个数 * 1048576` KiB。
```
Comment (qiancai) [inline check-before-deployment.md:637]:
```suggestion
> - 建议将 `vm.min_free_kbytes` 最小设置为 `1048576` KiB（即 1 GiB）。如果[安装了 NUMA](/check-before-deployment.md#安装-numactl-工具)，建议设置为 `NUMA 节点个数 * 1048576` KiB。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**